File: [0242][Easy][Valid_Anagram]/Valid_Anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s)!=len(t):
            return False
        else:
            #结果字典
            dic = {}
            #先遍历s，数每个字符出现的次数，存在字典dic中，key是字符，value是出现次数
            for i in s:
                if i in dic:
                    dic[i] +=1
                else:
                    dic[i] = 1
            #遍历t，遇见相同字符，就在dic中减一；遇见不同字符，直接返回false
            for i in t:
                if i in dic:
                    dic[i] -= 1
                else:
                    return False
            #遍历结果字典，是否每个value都为0
            for i in dic:
                if dic[i] != 0:
                    return False
        return True
    
    
def main():
    str1 = "anagram";
    str2 = "nagaram";
    Solution_obj = Solution()
    result = Solution_obj.isAnagram(str1, str2);
    print("In c code,result value is "+("FALSE" if result is False else "TRUE"));
    
if __name__ =='__main__':
    main()  

File: [0242][Easy][Valid_Anagram]/Valid_Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] alpha = new int[26];
        for (char a:s.toCharArray()) {
        	alpha[a-'a']++;
        }
        for (char a:t.toCharArray()) {
        	if (alpha[a-'a'] == 0)
        		return false;
        	else
        		alpha[a-'a'] --;
        }
        for (int i: alpha)
        	if (i != 0)
        		return false;
        return true;
    }
}

public class Valid_Anagram{
    public static void main(String args[]){
		String str1 = "anagram";
		String str2 = "nagaram";
		Solution Solution_obj = new Solution();
		boolean result = Solution_obj.isAnagram(str1, str2);
        System.out.println("In java code,last return value is :" + result);
    }
}